mm2src/mm2_main/src/lp_swap/maker_swap_v2.rs

```diff
@@ -968,6 +957,23 @@ impl<MakerCoin: MmCoin + MakerCoinSwapOpsV2, TakerCoin: MmCoin + TakerCoinSwapOp
             return Self::change_state(Aborted::new(reason), state_machine).await;
         }
 
+        let expected_taker_payment_fee = self.taker_payment_fee.amount;
+        let actual_taker_payment_fee =
+            big_decimal_from_sat_unsigned(taker_data.taker_payment_fee, state_machine.taker_coin.decimals());
+
+        // To avoid accidental division by zero, let's set the ratio to 1 if the expected fee is 0.
+        let ratio = if expected_taker_payment_fee == 0.into() {
+            1.into()
+        } else {
+            &actual_taker_payment_fee / &expected_taker_payment_fee
+        };
+
+        if ratio < 0.95.try_into().unwrap() {
```

Comment:
hmm I dont feel right to panic here, but if you think its alright, lets use `expect()`. At least it will have text message.

Also please use `from_str` for BigDecimal, it avoids the precision issues associated with floating-point numbers. Using a string representation like "0.95" avoids these issues because the string can be parsed directly into a BigDecimal without any intermediate floating-point representation.
![Screenshot 2024-05-20 at 13 53 03](https://github.com/KomodoPlatform/komodo-defi-framework/assets/52405288/70c3ef1c-1659-4163-a19f-186c6d6b19fb)

`if ratio < BigDecimal::from_str("0.95").expect("Failed to parse threshold value")`

Reply:
float conversions use string conversions under the hood, I checked that for when it might fail and that's only when the string is empty or has a non 10-base radix (so 0-9) (e.g. abcde in hex). Let's use `from_str` though since it is more consistent with what we are already using in the code base. And use an except detailing why this should never fail (I don't really think we need a new abort error type since a failure will never occur).